polkadot/scripts/list-syscalls/list-syscalls.rb

```diff
@@ -0,0 +1,608 @@
+#!/usr/bin/ruby
+
+# A script to statically list syscalls used by a given binary.
+#
+# Syntax:  list-syscalls.rb <binary> [--only-used-syscalls]
+#
+# NOTE:    For accurate results, build the binary with musl and LTO enabled.
+# Example: ./polkadot/scripts/list-syscalls/list-syscalls.rb target/x86_64-unknown-linux-musl/production/polkadot-prepare-worker --only-used-syscalls
+#
+# Author:  @koute
+# Source:  https://gist.github.com/koute/166f82bfee5e27324077891008fca6eb
+
+require 'shellwords'
+require 'set'
+
+SYNTAX_STRING = 'Syntax: list-syscalls.rb <binary> [--only-used-syscalls]'.freeze
+
+# Generated from `libc` using the following regex:
+#   'pub const SYS_([a-z0-9_]+): ::c_long = (\d+);'
+#   '    \2 => "\1",'
+SYSCALLS = {
```

Comment:
If we are going to commit syscall tables to the repo, I think we need a way of updating them.

We can simply have the upcoming CI test fail if the script finds a syscall number that is not in this map

Reply:
And if we're on this topic, are we going to also do this for arm? Do we know how many validators use arm machines?

Reply:
> If we are going to commit syscall tables to the repo, I think we need a way of updating them.

The script will just print out a number if it doesn't find the syscall in this table and will still work, so I think it's probably not worth to automate it and is fine to leave it as-is.

> And if we're on this topic, are we going to also do this for arm?

Nope.

> Do we know how many validators use arm machines?

Don't know about validators, but it's 0.5% of all nodes according to the telemetry.

Reply:
There was some talk of extending this to arm eventually. But yeah, it´s not on the radar just at this moment and not strongly needed. FYI, you can find telemetry [here](https://telemetry.polkadot.io/#stats/0x91b171bb158e2d3848fa23a9f1c25182fb8e20313b2c1eb49219da7a70ce90c3).

Reply:
> There was some talk of extending this to arm eventually. But yeah, it´s not on the radar just at this moment and not strongly needed. FYI, you can find telemetry [here](https://telemetry.polkadot.io/#stats/0x91b171bb158e2d3848fa23a9f1c25182fb8e20313b2c1eb49219da7a70ce90c3).

thanks for the link. Ok, makes sense for now.
Personally, I am hoping for a future where high-performance ARM servers will take over x86, but let's wait for that to become more realistic 😄